apis/proximity/src/tests.rs

```diff
@@ -0,0 +1,86 @@
+use core::cell::Cell;
+use libtock_platform::{share, ErrorCode, Syscalls, YieldNoWaitReturn};
+use libtock_unittest::fake;
+
+type Proximity = super::Proximity<fake::Syscalls>;
+
+#[test]
+fn no_driver() {
+    let _kernel = fake::Kernel::new();
+    assert_eq!(Proximity::exists(), Err(ErrorCode::NoDevice));
+}
+
+#[test]
+fn driver_check() {
+    let kernel = fake::Kernel::new();
+    let driver = fake::Proximity::new();
+    kernel.add_driver(&driver);
+
+    assert_eq!(Proximity::exists(), Ok(()));
+}
+
+#[test]
+fn busy_driver() {
+    let kernel = fake::Kernel::new();
+    let driver = fake::Proximity::new();
+    kernel.add_driver(&driver);
+
+    assert_eq!(Proximity::read(), Ok(()));
+    assert_eq!(Proximity::read(), Err(ErrorCode::Busy));
+    assert_eq!(Proximity::read_on_interrupt(0, 0), Err(ErrorCode::Busy));
+
+    driver.set_value(100);
+
+    assert_eq!(Proximity::read_on_interrupt(0, 0), Ok(()));
```

Comment:
Why is this `Ok(())` here?

Reply:
`driver.set_value(100)` executes a callback, consuming the read initiated by `read` at [line 28](https://github.com/alexandruCarp/libtock-rs/blob/7ce4c05c3fd2f4e718b3d4d6f9da42c897b8a447/apis/proximity/src/tests.rs#L28), so it is ok to initiate another measurement, be it on_interrupt or a normal read.

Reply:
Ok